solutionbox/structured_data/mltoolbox/_structured_data/preprocess/local_preprocess.py

```diff
@@ -92,20 +93,25 @@ def _init_numerical_results():
             'count': 0,
             'sum': 0.0}
   numerical_results = collections.defaultdict(_init_numerical_results)
-  categorical_results = collections.defaultdict(set)
+  categorical_results = collections.defaultdict(
+      lambda: collections.defaultdict(int))
 
   # for each file, update the numerical stats from that file, and update the set
   # of unique labels.
   for input_file in input_files:
     with file_io.FileIO(input_file, 'r') as f:
-      for line in f:
-        parsed_line = dict(zip(header, line.strip().split(',')))
+      for line in csv.reader(f):
+        parsed_line = dict(zip(header, line))
 
         for col_schema in schema_list:
           col_name = col_schema['name']
           col_type = col_schema['type']
           if col_type.lower() == 'string':
-            categorical_results[col_name].update([parsed_line[col_name]])
+            split_strings = parsed_line[col_name].split(' ')
```

Comment:
yea, I thought about this too, I don't want to be in the business of trying to fix the user's input. But I agree that this might be a common error. I'll make the change to filter ''.